source/include/shadow.h

```diff
@@ -342,94 +404,108 @@ typedef enum ShadowStatus
  * When thingName is only known at run time, do not use this macro. Use the
  * Shadow_GetTopicString() function instead.
  *
- * @param[operation] Can be one of:
- *                       - #SHADOW_OP_UPDATE
- *                       - #SHADOW_OP_DELETE
- *                       - #SHADOW_OP_GET
- * @param[suffix]    Can be one of:
- *                       - #SHADOW_SUFFIX_NULL
- *                       - #SHADOW_SUFFIX_ACCEPTED
- *                       - #SHADOW_SUFFIX_REJECTED
- *                       - #SHADOW_SUFFIX_DELTA
- *                       - #SHADOW_SUFFIX_DOCUMENTS
+ * @param[in] operation     Can be one of:
+ *                              - #SHADOW_OP_UPDATE
+ *                              - #SHADOW_OP_DELETE
+ *                              - #SHADOW_OP_GET
+ * @param[in] suffix        Can be one of:
+ *                              - #SHADOW_SUFFIX_NULL
+ *                              - #SHADOW_SUFFIX_ACCEPTED
+ *                              - #SHADOW_SUFFIX_REJECTED
+ *                              - #SHADOW_SUFFIX_DELTA
+ *                              - #SHADOW_SUFFIX_DOCUMENTS
  *
- * @param[thingName] Thing Name.
+ * @param[in] thingName     Thing Name.
+ * @param[in] shadowName    Shadow Name. Empty string for un-named ("Classic") shadow.
  *
  * @return Topic string.
  */
-#define SHADOW_TOPIC_STRING( thingName, operation, suffix ) \
-    ( SHADOW_PREFIX thingName operation suffix )
+#define SHADOW_TOPIC_STRING( thingName, shadowName, operation, suffix )           \
+    ( ( sizeof( shadowName ) > 1 ) ?                                              \
+      ( SHADOW_PREFIX thingName SHADOW_NAMED_ROOT shadowName operation suffix ) : \
+      ( SHADOW_PREFIX thingName SHADOW_CLASSIC_ROOT operation suffix ) )
 
 /**
- * @brief Assemble shadow topic string "$aws/things/<thingName>/shadow/update".
+ * @brief Assemble shadow topic string "$aws/things/<thingName>/shadow/update" or
+ *        "$aws/things/<thingName>/shadow/name/<shadowName>/update".
  */
-#define SHADOW_TOPIC_STRING_UPDATE( thingName ) \
-    SHADOW_TOPIC_STRING( thingName, SHADOW_OP_UPDATE, SHADOW_SUFFIX_NULL )
+#define SHADOW_TOPIC_STRING_UPDATE( thingName, shadowName ) \
+    SHADOW_TOPIC_STRING( thingName, shadowName, SHADOW_OP_UPDATE, SHADOW_SUFFIX_NULL )
```

Comment:
These topic string macros seem to be missing "/shadow/name/" for the shadowName case.